shared/lib-hoppy/hoppy/async_consumer.py

```diff
@@ -0,0 +1,210 @@
+import functools
+import logging
+import time
+from typing import Callable
+
+from hoppy.config import RABBITMQ_CONFIG
+from hoppy.hoppy_properties import ExchangeProperties, QueueProperties
+from hoppy.util import create_connection_parameters
+from pika.adapters.asyncio_connection import AsyncioConnection
+
+
+class AsyncConsumer(object):
+    def __init__(self,
+                 config: [dict | None] = None,
+                 exchange_properties: ExchangeProperties = ExchangeProperties(),
+                 queue_properties: QueueProperties = QueueProperties(),
+                 routing_key: str = '',
+                 reply_callback: Callable = None):
+        if config is None:
+            config = {}
+        self.config = {**RABBITMQ_CONFIG, **config}
+        self.connection_parameters = create_connection_parameters(self.config)
+
+        self.exchange_name = exchange_properties.name
+        self.exchange_type = exchange_properties.type
+        self.passive_declare_exchange = exchange_properties.passive_declare
+        self.durable_exchange = exchange_properties.durable
+        self.auto_delete_exchange = exchange_properties.auto_delete
+
+        self.queue_name = queue_properties.name
+        self.passive_declare_queue = queue_properties.passive_declare
+        self.durable_queue = queue_properties.durable
+        self.auto_delete_queue = queue_properties.auto_delete
+        self.exclusive_queue = queue_properties.exclusive
+
+        self.routing_key = routing_key
+
+        self._loop = None
+        self._connection = None
+        self._channel = None
+        self._consumer_tag = None
+        self._max_reconnect_delay = self.config.get('max_reconnect_delay', 30)
+
+        # In production, experiment with higher prefetch values
+        # for higher consumer throughput
+        self._prefetch_count = 1
```

Comment:
Magic number might be better as constant to provide better description and context